### DISCUSSION TITLE
Wrap long lines in diagnostic picker

### DISCUSSION THREAD

**Discussion by tsaraki** [helix-editor/helix#9785] [Ideas]
![image](https://github.com/helix-editor/helix/assets/21110604/2fe80f54-067b-4157-b198-1562ba9a5c9b)

to look like that:
![image](https://github.com/helix-editor/helix/assets/21110604/c842c0c5-afcc-446d-8a11-41df79d895b6)

**Comment by the-mikedavis** [comment]
IMO this is out of scope. The diagnostic picker is meant for jumping to errors rather than reading them. If you want to see a full diagnostic message you should jump to the diagnostic location

**Reply by tsaraki** [reply]
t is not so convenient to read the errors in the upper right corner, they are constantly mixed with other code.

The picker allows you to read the whole error, think, and jump to code or maybe go through the list of errors.

The changes affect various files, and a quick way to recognize the error to switch to necessary code section reduces programming time, which is why the picker is used for.

Also add something like this in the configuration:
[errors.show]
in_corner = false
in_picker = true

**Reply by willhansen** [reply]
It would be quite convenient to have a hotkey in the picker to cycle between "one line", "three lines", and "all lines" of diagnostic messages.

Understandable if not high priority though.

**Reply by zetashift** [reply]
> t is not so convenient to read the errors in the upper right corner, they are constantly mixed with other code.
This is so much, it's has happened a significant amount of time that the error overlaps the code, and then I can neither see the code or edit the code well enough to fix it.

I think inline diagnostics might help here, but even then a better UI for this might be possible using plugins.

**Reply by tsaraki** [reply]
@the-mikedavis is there any possibility to add wrap line in diagnostic picker in 24.03 due to update of Picker v2?

**Reply by the-mikedavis** [reply]
#9647 doesn't affect the table rendering code (https://github.com/helix-editor/helix/blob/master/helix-tui/src/widgets/table.rs) which is what would need to change to make this possible. I'm still skeptical that we need this anyways. It would complicate the table display (visually and codewise) and the diagnostic picker isn't meant to be an interface to read full diagnostic messages. #6417 will vastly improve the display of diagnostics

**Comment by tsaraki** [comment]
> IMO this is out of scope. The diagnostic picker is meant for jumping to errors rather than reading them. If you want to see a full diagnostic message you should jump to the diagnostic location

if picker is meant for jumping to next error, then why helix has [+D keymap?
Picker must be something else then just [+D
